ingestor/metrics/handler.go

```diff
@@ -163,3 +160,14 @@ func (s *Handler) HandleReceive(w http.ResponseWriter, r *http.Request) {
 	m.WithLabelValues(strconv.Itoa(http.StatusAccepted)).Inc()
 	w.WriteHeader(http.StatusAccepted)
 }
+
+func (s *Handler) Write(ctx context.Context, req *prompb.WriteRequest) error {
+	g, gCtx := errgroup.WithContext(ctx)
+	for _, writer := range s.requestWriters {
+		g.Go(func() error {
```

Comment:
I think the loop capture thing is fixed in Go 1.22 but worth verifying

```suggestion
		writer := writer
		g.Go(func() error {
```